crates/light-client-prover/src/da_block_handler.rs

```diff
@@ -149,10 +149,31 @@ where
 
         // Do any kind of ordering etc. on batch proofs here
         // If you do so, don't forget to do the same inside zk
+        let batch_prover_method_id = self
+            ._batch_proof_code_commitments_by_spec
+            .get(&SpecId::Genesis)
+            .expect("Batch proof code commitment not found");
+
+        let mut assumption = None;
+        let mut journal = None;
+        if !batch_proofs.is_empty() {
+            (assumption, journal) = match batch_proofs[0].clone() {
+                DaDataLightClient::Complete(proof) => {
+                    let data = proof.extract_proof();
+                    // Handle err
+                    let output = Vm::verify(&data, &batch_prover_method_id).unwrap();
+
+                    (Some(data), Some(output))
+                }
+                _ => (None, None),
+            };
+        }
 
-        let circuit_input = self.create_circuit_input(da_data, l1_block).await;
+        let circuit_input = self
```

Comment:
assumptions and journals should be vectors btw, multiple batch proofs on da can be used in the light client proof